crates/scroll/hardforks/src/hardfork.rs

```diff
@@ -0,0 +1,272 @@
+//! Hard forks of scroll protocol.
+
+use alloc::{boxed::Box, format, string::String, vec};
+use core::{
+    any::Any,
+    fmt::{self, Display, Formatter},
+    str::FromStr,
+};
+
+use alloy_chains::{Chain, NamedChain};
+use reth_ethereum_forks::{hardfork, ChainHardforks, EthereumHardfork, ForkCondition, Hardfork};
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+hardfork!(
+    /// The name of the Scroll hardfork
+    ///
+    /// When building a list of hardforks for a chain, it's still expected to mix with
+    /// [`EthereumHardfork`].
+    ScrollHardfork {
+        /// EIP150 HF block (nil = no fork)
+        EIP150Block,
+        /// EIP155 HF block
+        EIP155Block,
+        /// EIP158 HF block
+        EIP158Block,
```

Comment:
I think 150, 155 and 158 are part of [Tangerine](https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/tangerine-whistle.md) and [SpuriousDragon](https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/spurious-dragon.md). We can remove these.

You can compare the block activations from [here](https://ethereum.org/en/history) and [l2geth](https://github.com/scroll-tech/go-ethereum/blob/d2fe42ba8f4ed4c374d4e0c3b8b696760c601ed7/params/config.go#L70-L72) to see how they match.

Reply:
fixed [5aea8e8](https://github.com/scroll-tech/reth/pull/56/commits/5aea8e8f9a41db16b531d92a1995fc75e70fa828)